packages/markdown/processor/plugin/section-anchor-id.js

```diff
@@ -20,14 +20,29 @@ const getTexts = node => {
 function transformer(ast) {
   return flatMap(ast, node => {
     if (matchTag.test(node.tagName)) {
-      const id = `section-${kebabCase(getTexts(node))}`;
-      const element = {
+      // Parse the node texts to construct
+      // a backwards-compatible anchor ID.
+      const text = getTexts(node);
+      const id = `section-${kebabCase(text)}`;
+
+      if (node?.properties?.id) {
+        // Strip dash prefixes in GitHub slugger IDs
+        node.properties.id = node.properties.id.replace(/^(-+(?=\w))/, '');
```

Comment:
Should this also update `id` to match this? Doesn't seem right that `node.properties.id` wouldn't match the `properties.id` that's set on the anchor on line 39.

Reply:
Our new [slug generator](https://github.com/Flet/github-slugger) isn't an exact one-to-one with our old anchors, so we have to provide for backwards compatibility. We do that here by generating an additional anchor using our old, lodash-based strategy and injecting that in to the header HAST. In short these are intentionally different!